lib/events.js

```diff
@@ -54,64 +89,104 @@ $(document).ready(function() {
-                        meal.appendToTable($snackIndex)
-                        meal.mealCalories($snackCal)
-                        meal.mealCaloriesRemaining($snackRemainingCal, bfastTarget)
+                        meal.populateTable($snackIndex, $snackCal, $snackGoalCal, $snackRemainingCal, snackTarget)
                         break;
                     case "Dinner":
-                        meal.appendToTable($dinnerIndex)
-                        meal.mealCalories($dinnerCal)
-                        meal.mealCaloriesRemaining($dinnerRemainingCal, bfastTarget)
+                        meal.populateTable($dinnerIndex, $dinnerCal, $dinnerGoalCal, $dinnerRemainingCal, dinnerTarget)
                         break;
                 }
             })
             return meals
         })
-        .then(function(meals) {
-            let total = meals.reduce(function(sum, meal) {
-                return sum + meal.total
-            }, 0)
-            let remaining = target - total
-            $totalCal.text(total)
-            $remainingCal.text(remaining)
-            $goalCal.text(target)
-            if (remaining < 0) {
-                $remainingCal.addClass('negative')
-            } else {
-                $remainingCal.removeClass('negative')
-            }
+        .then(function() {
+            Meal.updateGrandTotal(target)
         })
 
     $foodSubmit.on('click', function() {
         let food = {
             name: $newFoodName.val(),
             calories: $newFoodCalories.val()
         }
-        if(Food.validate(food)) {
-          Food.addNewFood(food, $foodIndex);
-          $newFoodName.val('');
-          $newFoodCalories.val('');
-          $('.notify').remove();
+
+        if (Food.validate(food)) {
+            Food.addNew(food, $foodIndex);
+            $newFoodName.val('');
+            $newFoodCalories.val('');
+            $('.notify').remove();
         }
     })
 
     $foodIndex.on('click', '.delete', function(event) {
         var food = this.parentElement.parentElement
         var id = food.id
-        Food.deleteFood(id)
+        Food.delete(id)
             .then(function() {
                 $(`#${id}`).fadeOut(300);
             })
     })
 
-})
+    $mealButtons.on('click', function(event) {
+        let meal = event.target.id
+        let $mealTable = $(`#${meal}-items`)
+
+        $('input:checked').each(function(key, food) {
+            let node = food.closest('tr')
+            let foodItem = {
+                id: node.id,
+                name: node.children[0].innerText,
+                calories: node.children[1].innerText
+            }
+            Meal.addFoodItem($mealTable.data().id, foodItem.id)
+                .then(function(data) {
+                    let foodObj = new Food(foodItem)
+                    $mealTable.prepend(foodObj.toHTML())
+                    Meal.updateTotal($mealTable)
+                })
+        })
+        $('input:checkbox:checked').prop('checked', false);
```

Comment:
same here. could you scope this further by leveraging what `this`, `currentTarget` or `target` is?